src/libopenrave/kinbody.cpp

```diff
@@ -726,16 +722,64 @@ void KinBody::SetLinkGroupGeometries(const std::string& geomname, const std::vec
     _PostprocessChangedParameters(Prop_LinkGeometryGroup); // have to notify collision checkers that the geometry info they are caching could have changed.
 }
 
+void KinBody::_InitLinkFromInfo(KinBody::LinkPtr& linkPtr, const KinBody::LinkInfo& linkInfo)
+{
+    linkPtr->_vGeometries.clear();
+    linkPtr->_collision.vertices.clear();
+    linkPtr->_collision.indices.clear();
```

Comment:
we might know in advance how many items should be there, might "reserve()" instead of adding elements one-by-one with reallocations

Reply:
added reserve for _vGeometries, but the collision vectors are sized based on the mesh vertices, which aren't known yet / probably aren't 1:1?

Reply:
agree about collisions, thank you!